src/test/scala/uk/gov/hmrc/play/microservice/bootstrap/JsonErrorHandlingSpec.scala

```diff
@@ -51,6 +55,35 @@ class JsonErrorHandlingSpec extends WordSpecLike with Matchers with ScalaFutures
       resultF.header.status shouldBe 500
     }
 
+    "log an error for a PlayException" in {
+      withCaptureOfLoggingFrom(Logger) { logEvents =>
+        val method = "some-method"
+        val uri = "some-uri"
+        when(requestHeader.method).thenReturn(method)
+        when(requestHeader.uri).thenReturn(uri)
 
+        jsh.onError(requestHeader, UnexpectedException(Some("any application exception"))).futureValue
+
+        eventually {
+          val message = logEvents.filter(_.getLevel == Level.ERROR).loneElement.getMessage
+          message should include regex s"! @[a-zA-Z0-9]+ - Internal server error, for \\($method\\) \\[$uri\\] ->".r
+        }
+      }
+    }
+
+    "log an error" in {
+      withCaptureOfLoggingFrom(Logger) { logEvents =>
+        val method = "some-method"
+        val uri = "some-uri"
+        when(requestHeader.method).thenReturn(method)
+        when(requestHeader.uri).thenReturn(uri)
+
+        jsh.onError(requestHeader, new Exception("any application exception")).futureValue
+
+        eventually {
+          logEvents.filter(_.getLevel == Level.ERROR).loneElement.getMessage should include(s"! Internal server error, for ($method) [$uri] ->")
```

Comment:
With this statement you probably want to make sure the _only_ error log recorded by the test appender is the one with:
- the expected level (ERROR)
- the expected message
So I don't see the point of filtering logEvents.
Also, the assertion on the log message could be stronger (shouldBe).